src/pages/member/QuickRegister.js

```diff
@@ -0,0 +1,247 @@
+                <Typography variant='h2' className={classes.header}>
+                  {event.ename}
+                </Typography>
+                <Typography className={classes.subHeader}>
+            Sign up Form
+                </Typography>
+                <Formik
+                  initialValues={initialValues}
+                  validationSchema={validationSchema}
+                  onSubmit={submitValues}>
+                  {props => <RegisterQuick {...props} />}
+                </Formik>
+              </div>
+            </Paper>
+          </div>}
+      </React.Fragment>
+    )
+  } else {
+    return (
+      <div className={classes.layout}>
+        <Paper className={classes.paper}>
+          <Skeleton animation='wave' variant='rect' width={'100%'} height={320} />
+          <div className={classes.content}>
+
+            <Grid container spacing={3}>
+
+              <Grid item xs={12}>
+                <Skeleton animation='wave' variant='rect' width={300} height={30} />
+              </Grid>
+
+              {[1, 2, 3].map((e) =>
+                <Grid item container spacing={1} key={e}>
+                  <Grid item xs={12}>
+                    <Skeleton animation='wave' variant='rect' width={130} height={20} />
+                  </Grid>
+                  <Grid item xs={12}>
+                    <Skeleton animation='wave' variant='rect' width={'100%'} height={20} />
+                  </Grid>
+                </Grid>)
+              }
+
+              <Grid item xs={12}>
+                <Skeleton animation='wave' variant='rect' width={90} height={36} />
+              </Grid>
+
+            </Grid>
+          </div>
+        </Paper>
+      </div>
+    )
+  }
+
+  async function submitValues (values) {
+    const { email, fname, lname, id, faculty, year, diet, heardFrom, gender } = values
+    const eventID = event.id
+    // TODO: Standardize the values passed to DB (right now it passes "1st Year" instead of 1)
+    const body = {
+      id,
+      fname,
+      lname,
+      email,
+      year,
+      faculty,
+      gender,
+      diet
+    }
+    fetchBackend(`/users/${values.id}`, 'GET')
+      .then(() => {
+        // if get response is successful
+        fetchBackend(`/users/${id}`, 'PATCH', body)
+        registerUser(id, eventID, heardFrom)
+      })
+      .catch(() => {
+        // Need to create new user
+        fetchBackend('/users', 'POST', body)
+          .then((userResponse) => {
+            if (userResponse.message === 'Created!') {
+              registerUser(id, eventID, heardFrom)
+            } else {
+              alert('Signup failed')
+            }
+          })
+      })
+  }
+
+  async function registerUser (id, eventID, heardFrom) {
+    const body = {
+      id,
+      eventID,
+      heardFrom,
+      registrationStatus: 'registered'
+    }
+    fetchBackend('/registrations', 'POST', body)
+      .then((regResponse) => {
+        alert('Signed Up')
```

Comment:
Do we need this if they see the Done screen afterwards?

Reply:
good point